checkov/terraform/module_loading/module_finder.py

```diff
@@ -36,6 +36,22 @@ def __str__(self) -> str:
 def find_modules(path: str) -> List[ModuleDownload]:
     modules_found: list[ModuleDownload] = []
 
+    # Leverage modules.json to better inform discovery. If we have this,
+    # there should be no need to walk and gather modules
+    if env_vars_config.CHECKOV_EXPERIMENTAL_TERRAFORM_MANAGED_MODULES:
```

Comment:
Please consider extracting this logic into a separate function for better readability and reusability